File: 02_flashlight/c/flashlight.c
#include <wiringPi.h>
#include <stdio.h>

#define ledPin 0
#define buttonPin 1

void main(void)
{
	/* Required to initialize WiringPi */
	wiringPiSetup();

	pinMode(ledPin, OUTPUT);
	printf("Pin #%d set to OUTPUT mode.\n", ledPin);

	pinMode(buttonPin, INPUT);
	pullUpDnControl(buttonPin, PUD_UP);
	printf("Pin #%d set to INPUT mode (pull-up).\n", buttonPin);

	while (TRUE)
	{
		if (digitalRead(buttonPin) == LOW)
		{
			printf("Turning LED on...\n");
			digitalWrite(ledPin, HIGH);
		}
		else
		{
			printf("Turning LED off...\n");
			digitalWrite(ledPin, LOW);
		}
	}
}

File: 02_flashlight/python/flashlight.py
import RPi.GPIO as GPIO

ledPin = 11
buttonPin = 12


def setup():
    GPIO.setmode(GPIO.BOARD)

    GPIO.setup(ledPin, GPIO.OUT)
    GPIO.output(ledPin, GPIO.LOW)
    print('Pin #%d set to OUTPUT mode.' % ledPin)

    GPIO.setup(buttonPin, GPIO.IN, pull_up_down=GPIO.PUD_UP)
    print('Pin #%d set to INPUT mode (pull-up).' % buttonPin)


def loop():
    while True:
        if GPIO.input(buttonPin) == GPIO.LOW:
            print('Turning LED on...')
            GPIO.output(ledPin, GPIO.HIGH)
        else:
            print('Turning LED on...')
            GPIO.output(ledPin, GPIO.LOW)


def destroy():
    GPIO.output(ledPin, GPIO.LOW)
    GPIO.cleanup()


if __name__ == '__main__':
    setup()

    try:
        loop()
    except KeyboardInterrupt:
        destroy()
